misc-scripts/xref_mapping/XrefParser/ProcessData.pm

```diff
@@ -682,26 +684,21 @@ sub dbi {
 ###########################################################
 sub validate_species {
   my ($self, $species, $verbose) = @_;
-  my @species_ids;
 
   my $dbi = $self->dbi();
   my $sth = $dbi->prepare("SELECT species_id, name FROM species WHERE LOWER(name)=? OR LOWER(aliases) REGEXP ?");
   my ($species_id, $species_name);
 
-  foreach my $sp (@$species) {
-
-    my $bind_arg = "^".lc($sp).",|^".lc($sp)."\$|,[ ]{0,1}".lc($sp)."[ ]{0,1},|,[ ]{0,1}".lc($sp)."\$";
-    $sth->execute(lc($sp), $bind_arg ); 
-    $sth->bind_columns(\$species_id, \$species_name);
-    if (my @row = $sth->fetchrow_array()) {
-      print "Species $sp is valid (name = " . $species_name . ", ID = " . $species_id . ")\n" if($verbose);
-      push @species_ids, $species_id;
-    } else {
-      print STDERR "Species $sp is not valid; valid species are:\n";
-      $self->show_valid_species();
-    }
+  my $bind_arg = "^".lc($species).",|^".lc($species)."\$|,[ ]{0,1}".lc($species)."[ ]{0,1},|,[ ]{0,1}".lc($species)."\$";
+  $sth->execute(lc($species), $bind_arg ); 
+  $sth->bind_columns(\$species_id, \$species_name);
+  if (my @row = $sth->fetchrow_array()) {
+    print "Species $species is valid (name = " . $species_name . ", ID = " . $species_id . ")\n" if($verbose);
+  } else {
```

Comment:
If block has just the print statement?!
Where is the $species_id getting initialised?!!
Probably the if block is missing $species_id initialisation?!!!

Reply:
the fetchrow_array method populates the species_id and species_name values